### DISCUSSION TITLE
unauthorized error accessing api channel

### DISCUSSION THREAD

**Discussion by rpg101a** [nextgenhealthcare/connect#4901] [Q&A]
I'm fairly new to Mirth. Trying to write a program to monitor when a channel fails or stops, but getting a:

**HTTP ERROR 401
Problem accessing /api/channels/idsAndNames. 
_Reason:
    Unauthorized_

I have admin rights. Do I need to add my mirth u/p to the url? The doc doesn't mention anything, so I'm wondering what is causing this

Thank you

**Comment by pacmano1** [comment]
Did you credential to the endpoint?    If you were using a mirth http sender:

![image](https://user-images.githubusercontent.com/44065187/145678896-487a6271-a2ff-4133-a04b-6d22feabe2b9.png)
